File: backend/app/Models/WashService.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class WashService extends Model
{
    protected $fillable = [
        'vehicle_type_id',
        'name',
        'base_price',
        'is_custom',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'base_price' => 'decimal:2',
            'is_custom'  => 'boolean',
            'is_active'  => 'boolean',
        ];
    }

    public function vehicleType(): BelongsTo
    {
        return $this->belongsTo(VehicleType::class);
    }

    public function washes(): HasMany
    {
        return $this->hasMany(Wash::class);
    }
}

File: carwash_app/lib/features/washes/models/wash_service.dart
class WashService {
  final int id;
  final int vehicleTypeId;
  final String name;
  final double? basePrice;
  final bool isCustom;
  final bool isActive;

  const WashService({
    required this.id,
    required this.vehicleTypeId,
    required this.name,
    this.basePrice,
    required this.isCustom,
    required this.isActive,
  });

  factory WashService.fromJson(Map<String, dynamic> json) {
    return WashService(
      id: json['id'] as int,
      vehicleTypeId: json['vehicle_type_id'] as int,
      name: json['name'] as String,
      basePrice: json['base_price'] != null
          ? double.parse(json['base_price'].toString())
          : null,
      isCustom: json['is_custom'] as bool,
      isActive: json['is_active'] as bool,
    );
  }
}
